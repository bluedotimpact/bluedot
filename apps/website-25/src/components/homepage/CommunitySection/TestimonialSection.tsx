import { Section } from '@bluedot/ui';
import { SlideList } from '@bluedot/ui/src/SlideList';

type Testimonial = {
  quote: string;
  name: string;
  role: string;
  imageSrc: string;
};

const testimonials: Testimonial[] = [
  {
    quote: "This was the most positively impactful course I've ever taken (unless you count the high school class in which I got to know my husband!), as it gave me the background to engage with the AI safety and governance communities. I don't know how I would have gotten up to speed otherwise, and it opened the door to pretty much everything I've done professionally for the past couple years.",
    name: 'Kendrea Beers',
    role: 'Horizon Junior Fellow at the Center for Security and Emerging Technology',
    imageSrc: '/images/graduates/kendrea.png',
  },
  {
    quote: "I participated in the AISF Alignment Course last year and consider it to be the single most useful step I've taken in my career so far. I cannot recommend the program strongly enough",
    name: 'Sarah Cogan',
    role: 'Software Engineer at Google DeepMind',
    imageSrc: '/images/graduates/sarah.png',
  },
  {
    quote: "This course was the first step in my career in AI safety. The BlueDot course allowed me to bridge the gap between my previous career as an economist to now working in the UK Government AI Directorate. The course provided me with a great introduction to the field, and allowed me to meet some great people with whom I am still friends. I'd recommend this course to anyone!",
    name: 'Matthew Bradbury',
    role: 'Senior AI Risk Analyst in the UK Government',
    imageSrc: '/images/graduates/matthew.png',
  },
];

const TestimonialSection = () => {
  return (
    <Section>
      <SlideList
        subtitle="What our graduates say"
        maxItemsPerSlide={3}
        slideClassName="px-2"
        containerClassName="justify-center"
      >
        {testimonials.map((testimonial) => (
          <div key={testimonial.name} className="testimonial flex flex-col h-full border rounded-radius-md p-8">
            <blockquote className="testimonial__quote text-size-l text-bluedot-darker mb-8 flex-grow text-center">
              "{testimonial.quote}"
            </blockquote>
            <div className="testimonial__footer flex items-center gap-4 mt-auto">
              <div className="testimonial__avatar size-16 rounded-full overflow-hidden flex-shrink-0">
                <img
                  src={testimonial.imageSrc}
                  alt={testimonial.name}
                  className="size-full object-cover"
                />
              </div>
              <div className="testimonial__info">
                <div className="testimonial__name font-semibold text-bluedot-black">
                  {testimonial.name}
                </div>
                <div className="testimonial__role text-size-s text-bluedot-darker">
                  {testimonial.role}
                </div>
              </div>
            </div>
          </div>
        ))}
      </SlideList>
    </Section>
  );
};

export default TestimonialSection;
