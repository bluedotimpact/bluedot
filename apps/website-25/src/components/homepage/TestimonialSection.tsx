import Image from 'next/image';

interface Testimonial {
  quote: string;
  name: string;
  role: string;
  image: string;
}

const testimonials: Testimonial[] = [
  {
    quote: "This was the most positively impactful course I've ever taken (unless you count the high school class in which I got to know my husband!), as it gave me the background to engage with the AI safety and governance communities. I don't know how I would have gotten up to speed otherwise, and it opened the door to pretty much everything I've done professionally for the past couple years.",
    name: "Kendrea Beers",
    role: "Horizon Junior Fellow at the Center for Security and Emerging Technology",
    image: "/images/graduates/kendrea.png"
  },
  {
    quote: "I participated in the AISF Alignment Course last year and consider it to be the single most useful step I've taken in my career so far. I cannot recommend the program strongly enough",
    name: "Sarah Cogan",
    role: "Software Engineer at Google DeepMind",
    image: "/images/graduates/sarah.png"
  },
  {
    quote: "This course was the first step in my career in AI safety. The BlueDot course allowed me to bridge the gap between my previous career as an economist to now working in the UK Government AI Directorate. The course provided me with a great introduction to the field, and allowed me to meet some great people with whom I am still friends. I'd recommend this course to anyone!",
    name: "Matthew Bradbury",
    role: "Senior AI Risk Analyst in the UK Government",
    image: "/images/graduates/matthew.png"
  }
];

const TestimonialSection = () => {
  return (
    <section className="testimonial-section py-24">
      <div className="testimonial-section__container container-lined mx-auto max-w-[1750px]">
        <h2 className="testimonial-section__title text-2xl font-bold text-bluedot-darker mb-12 px-6 pt-6">
          What our graduates say
        </h2>
        
        <div className="testimonial-section__grid grid grid-cols-1 md:grid-cols-3 gap-[30px] divide-x divide-gray-200 px-[46px] pb-[78px] text-black">
          {testimonials.map((testimonial, index) => (
            <div key={index} className="testimonial-section__card flex flex-col px-8 first:pl-0 last:pr-0">
              <blockquote className="testimonial-section__quote text-lg mb-6 line-clamp-[8]">
                "{testimonial.quote}"
              </blockquote>
              <div className="testimonial-section__profile flex items-center gap-4 mt-auto">
                <div className="testimonial-section__image-wrapper w-16 h-16 relative rounded-full overflow-hidden">
                  <Image
                    src={testimonial.image}
                    alt={testimonial.name}
                    fill
                    className="testimonial-section__image object-cover"
                  />
                </div>
                <div className="testimonial-section__profile-info">
                  <div className="testimonial-section__name font-semibold">
                    {testimonial.name}
                  </div>
                  <div className="testimonial-section__role text-black">
                    {testimonial.role}
                  </div>
                </div>
              </div>
            </div>
          ))}
        </div>
      </div>
    </section>
  );
};

export default TestimonialSection;

