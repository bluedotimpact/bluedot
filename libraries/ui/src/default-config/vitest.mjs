/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
