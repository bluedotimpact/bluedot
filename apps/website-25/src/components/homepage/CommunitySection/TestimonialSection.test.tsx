import { describe, expect, test } from 'vitest';
import { render, screen } from '@testing-library/react';
import TestimonialSection from './TestimonialSection';

describe('TestimonialSection', () => {
  test('renders as expected', () => {
    const { container } = render(<TestimonialSection />);
    expect(container).toMatchSnapshot();
  });

  test('displays testimonials not in a slide list', () => {
    render(<TestimonialSection />);

    // Check for testimonial content
    const testimonials = screen.getAllByRole('blockquote');
    expect(testimonials).toHaveLength(3);

    // Verify slide navigation does not exist
    expect(screen.queryByLabelText('Previous slide')).toBeNull();
    expect(screen.queryByLabelText('Next slide')).toBeNull();
  });

  test('displays section title', () => {
    render(<TestimonialSection />);
    expect(screen.getByText(/What our graduates say/i)).toBeDefined();
  });
});
