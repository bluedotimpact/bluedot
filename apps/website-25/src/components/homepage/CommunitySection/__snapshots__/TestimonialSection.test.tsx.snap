// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TestimonialSection > renders as expected 1`] = `
<div>
  <div
    class="section w-full max-w-max-width min-w-min-width mx-auto overflow-y-scroll md:px-gutter px-gutter-sm my-8"
  >
    <div
      class="section__title-container flex justify-between items-center gap-4"
    >
      <div
        class="section__content flex-1"
      />
    </div>
    <div
      class="section__body"
    >
      <section
        class="slide-list w-full"
      >
        <div
          class="slide-list__header flex justify-between items-start mb-8"
        >
          <div
            class="slide-list__header-content"
          >
            <h2
              class="slide-list__title text-4xl font-bold text-bluedot-darker"
            >
              What our graduates say
            </h2>
          </div>
        </div>
        <div
          class="slide-list__content flex gap-2 justify-center"
        >
          <div
            class="slide-list__container relative overflow-hidden flex-shrink-0"
            style="width: 100%;"
          >
            <div
              class="slide-list__track flex transition-transform duration-300 ease-in-out pb-8"
              style="transform: translateX(-0%);"
            >
              <div
                class="slide-list__track-item w-full flex-shrink-0 flex gap-2 gap-8"
              >
                <div
                  class="slide-list__item flex-1"
                >
                  <div
                    class="size-full"
                  >
                    <div
                      class="testimonial flex flex-col h-full border rounded-lg p-8"
                    >
                      <blockquote
                        class="testimonial__quote text-lg text-bluedot-darker mb-8 flex-grow text-center"
                      >
                        "
                        This was the most positively impactful course I've ever taken (unless you count the high school class in which I got to know my husband!), as it gave me the background to engage with the AI safety and governance communities. I don't know how I would have gotten up to speed otherwise, and it opened the door to pretty much everything I've done professionally for the past couple years.
                        "
                      </blockquote>
                      <div
                        class="testimonial__footer flex items-center gap-4 mt-auto"
                      >
                        <div
                          class="testimonial__avatar size-16 rounded-full overflow-hidden flex-shrink-0"
                        >
                          <img
                            alt="Kendrea Beers"
                            class="size-full object-cover"
                            src="/images/graduates/kendrea.png"
                          />
                        </div>
                        <div
                          class="testimonial__info"
                        >
                          <div
                            class="testimonial__name font-semibold text-bluedot-black"
                          >
                            Kendrea Beers
                          </div>
                          <div
                            class="testimonial__role text-sm text-bluedot-darker"
                          >
                            Horizon Junior Fellow at the Center for Security and Emerging Technology
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="slide-list__item flex-1"
                >
                  <div
                    class="size-full"
                  >
                    <div
                      class="testimonial flex flex-col h-full border rounded-lg p-8"
                    >
                      <blockquote
                        class="testimonial__quote text-lg text-bluedot-darker mb-8 flex-grow text-center"
                      >
                        "
                        I participated in the AISF Alignment Course last year and consider it to be the single most useful step I've taken in my career so far. I cannot recommend the program strongly enough
                        "
                      </blockquote>
                      <div
                        class="testimonial__footer flex items-center gap-4 mt-auto"
                      >
                        <div
                          class="testimonial__avatar size-16 rounded-full overflow-hidden flex-shrink-0"
                        >
                          <img
                            alt="Sarah Cogan"
                            class="size-full object-cover"
                            src="/images/graduates/sarah.png"
                          />
                        </div>
                        <div
                          class="testimonial__info"
                        >
                          <div
                            class="testimonial__name font-semibold text-bluedot-black"
                          >
                            Sarah Cogan
                          </div>
                          <div
                            class="testimonial__role text-sm text-bluedot-darker"
                          >
                            Software Engineer at Google DeepMind
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="slide-list__item flex-1"
                >
                  <div
                    class="size-full"
                  >
                    <div
                      class="testimonial flex flex-col h-full border rounded-lg p-8"
                    >
                      <blockquote
                        class="testimonial__quote text-lg text-bluedot-darker mb-8 flex-grow text-center"
                      >
                        "
                        This course was the first step in my career in AI safety. The BlueDot course allowed me to bridge the gap between my previous career as an economist to now working in the UK Government AI Directorate. The course provided me with a great introduction to the field, and allowed me to meet some great people with whom I am still friends. I'd recommend this course to anyone!
                        "
                      </blockquote>
                      <div
                        class="testimonial__footer flex items-center gap-4 mt-auto"
                      >
                        <div
                          class="testimonial__avatar size-16 rounded-full overflow-hidden flex-shrink-0"
                        >
                          <img
                            alt="Matthew Bradbury"
                            class="size-full object-cover"
                            src="/images/graduates/matthew.png"
                          />
                        </div>
                        <div
                          class="testimonial__info"
                        >
                          <div
                            class="testimonial__name font-semibold text-bluedot-black"
                          >
                            Matthew Bradbury
                          </div>
                          <div
                            class="testimonial__role text-sm text-bluedot-darker"
                          >
                            Senior AI Risk Analyst in the UK Government
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
